Closes #156 Support discount codes

nikola-mladenovic opened the pull request:
Even though we'll change the discount codes UI with the redesign, this is what I currently came up with:

- To get the information on the currently applied discounts, a customer should tap on the _Order Discount_ information icon in the summary section of the cart screen;
- Currently proposed (and implemented) textual representation for the discount codes:
  - For every discount code (_[]_ are not included in the textual representation, just indicating that a segment is optional):
  - _actual discount code_  [: _discount code name_] [(_discount code description_)] 
  - _new line_
    - Now, for every cart discount that belongs to this discount code:
    - _cart discount name_  [(_discount code description_)] [_permyriad / 100_ %] [_money_]
- To add a new discount code, user can tap on the button and we'll show an error message, or just apply it and show the updated UI.

If you look into the changed files, you'll also be able to see the German translations I provided using google translator for the newly introduced strings which are in English. Any feedback more than welcome.

TestFlight build from this branch is # 95.

nkuehn commented on Sunrise/de.lproj/Localizable.strings:
```diff
@@ -175,4 +175,16 @@
 "Get Directions" = "Anfahrt";
 
+/* Discounts */
+"Discounts" = "Rabatt";
```
"Discounts" is plural, "Rabatt" is singular.  Which do you want?    Plural in German in "Rabatte"

Keep in mind that a "Discount Code" is not a "Discount" .   A Discount Code is usually called "Gutscheincode" or "Coupon"  (like in english, often called "Coupon" there, too).

nikola-mladenovic replied:
This should be the title of the dialogue showing textual representation of all the discounts which have been applied. So IMHO I'd use plural (Discounts or Coupons).

nkuehn commented on Sunrise/de.lproj/Localizable.strings:
```diff
@@ -175,4 +175,16 @@
+"Discounts" = "Rabatt";
+
+/* Add discount code message */
+"Add your discount code bellow:" = "Fügen Sie Ihren Rabatt Code unten hinzu:";
```
"bellow"  has only one "l" in english:  "below" is correct. 

"Rabatt Code" would usually be made a compound word in geman  ->  "Rabattcode"  or "Gutscheincode" or "Coupon".

nkuehn replied:
btw: why do you need "below" at all?   If the input is adjacent this is just noise to me - does not matter if it's directly below or directly next to it,  a colon ":"  should suffice IMHO

nikola-mladenovic replied:
I agree - we can remove that _noise_, it was something I wrote at that moment, it's not very useful..

Can you give me the translation for _Add your discount code:_ ?

nkuehn replied:
sure-  "Rabattcode hinzufügen:"   (German UIs tend to not use full sentences since the sentence structure is more verbose than the english one - english literal translation would be "Add Discount Code:" )

nkuehn commented on Sunrise/de.lproj/Localizable.strings:
```diff
@@ -175,4 +175,16 @@
+"Discount" = "Rabatt";
+
+/* No discounts */
+"No discounts - please redeem a code." = "Keine Rabatte - bitte erlöschen Sie einen Code.";
```
FYI: there can be discounts applied to the cart without a code being added.   I haven't been able try due to lack of data in sunrise (or I don't know what to add to the cart), but you may want to check with @mmoelli  on the wording and logic.  
Discount Codes are just an optional extra condition for a cart discount.

nkuehn replied:
"erlöschen" is completely off - maybe we shouldn't use google translate for UI text :-)   (it rather means "expire"). 

You probaby look for "einlösen" here.   -> "Keine Rabatte - bitte lösen sie einen Code ein."

nikola-mladenovic replied:
Thanks for the translation.

> there can be discounts applied to the cart without a code being added.

Let's rephrase and say - _You haven't added any discount codes to your cart._ WDYT?

nkuehn replied:
sounds good if you want to use the space and actually tell it.   
If it's the cart details view where you can't edit anything anyways I personally would go for more minimalism and just not mention the code at all if there are none applied.  Since no discounted price is shown, too then there is no need to explain that there is no code applied.

nikola-mladenovic replied:
Feel free to check out the build - this is actually shown in the popup presented when you tap on ℹ️ icon in the cart summary (discount section).

nkuehn replied:
right, thanks for the tip - I didn't see that this PR is already in the build.

nkuehn replied:
just tried - sorry to say, I think you need to rethink the concept more generally. 

Try to put M0E20000000EAJR   in the cart ("Dress ana Alcazar multi" in size 34).  It has a cart discount applied that does not require a code.  So you see a Discount line but if you click the info icon it shows the error message. 

If I click the info icon next to a discount amount I want to see the name of the discount / how the discounts are mad up of multiple discount promotions and not an error that no code is applied.

nikola-mladenovic replied:
Good point 👍 

Since the UX will likely change with the new design, I'd try to fix this by removing the ℹ️ icon when there are no discount codes present. Does that sound like an OK approach for the time being?

nkuehn replied:
sounds like a good approach to make it consistent without refactoring!

nikola-mladenovic commented:
Latest commit addresses things discussed above, check it out in the TestFlight build # 96.

One more thing I'm interested to hear your opinion is the textual representation for the discount codes array I proposed above (and implemented in the alert view / popup you can see on the cart screen).

nikola-mladenovic commented:
To avoid opening more PRs before Christoph gets a chance to review the existing ones, I've added a couple of commits to this branch:
- For my store and store selection screens, we use user's location, to show the distance from the available stores. Since we don't need to track every single movement / change, I've modified the code to track significant location changes, and save some battery.
- Removed _MRS._ option for the customer sign up (issue #151 ).
